src/color_picker.coffee

```diff
@@ -1,258 +1,84 @@
-  gainsboro: "#dcdcdc"
-  ghostwhite: "#f8f8ff"
-  gold: "#ffd700"
-  goldenrod: "#daa520"
-  gray: "#808080"
-  green: "#008000"
-  greenyellow: "#adff2f"
-  honeydew: "#f0fff0"
-  hotpink: "#ff69b4"
-  "indianred ": "#cd5c5c"
-  indigo: "#4b0082"
-  ivory: "#fffff0"
-  khaki: "#f0e68c"
-  lavender: "#e6e6fa"
-  lavenderblush: "#fff0f5"
-  lawngreen: "#7cfc00"
-  lemonchiffon: "#fffacd"
-  lightblue: "#add8e6"
-  lightcoral: "#f08080"
-  lightcyan: "#e0ffff"
-  lightgoldenrodyellow: "#fafad2"
-  lightgrey: "#d3d3d3"
-  lightgreen: "#90ee90"
-  lightpink: "#ffb6c1"
-  lightsalmon: "#ffa07a"
-  lightseagreen: "#20b2aa"
-  lightskyblue: "#87cefa"
-  lightslategray: "#778899"
-  lightsteelblue: "#b0c4de"
-  lightyellow: "#ffffe0"
-  lime: "#00ff00"
-  limegreen: "#32cd32"
-  linen: "#faf0e6"
-  magenta: "#ff00ff"
-  maroon: "#800000"
-  mediumaquamarine: "#66cdaa"
-  mediumblue: "#0000cd"
-  mediumorchid: "#ba55d3"
-  mediumpurple: "#9370d8"
-  mediumseagreen: "#3cb371"
-  mediumslateblue: "#7b68ee"
-  mediumspringgreen: "#00fa9a"
-  mediumturquoise: "#48d1cc"
-  mediumvioletred: "#c71585"
-  midnightblue: "#191970"
-  mintcream: "#f5fffa"
-  mistyrose: "#ffe4e1"
-  moccasin: "#ffe4b5"
-  navajowhite: "#ffdead"
-  navy: "#000080"
-  oldlace: "#fdf5e6"
-  olive: "#808000"
-  olivedrab: "#6b8e23"
-  orange: "#ffa500"
-  orangered: "#ff4500"
-  orchid: "#da70d6"
-  palegoldenrod: "#eee8aa"
-  palegreen: "#98fb98"
-  paleturquoise: "#afeeee"
-  palevioletred: "#d87093"
-  papayawhip: "#ffefd5"
-  peachpuff: "#ffdab9"
-  peru: "#cd853f"
-  pink: "#ffc0cb"
-  plum: "#dda0dd"
-  powderblue: "#b0e0e6"
-  purple: "#800080"
-  red: "#ff0000"
-  rosybrown: "#bc8f8f"
-  royalblue: "#4169e1"
-  saddlebrown: "#8b4513"
-  salmon: "#fa8072"
-  sandybrown: "#f4a460"
-  seagreen: "#2e8b57"
-  seashell: "#fff5ee"
-  sienna: "#a0522d"
-  silver: "#c0c0c0"
-  skyblue: "#87ceeb"
-  slateblue: "#6a5acd"
-  slategray: "#708090"
-  snow: "#fffafa"
-  springgreen: "#00ff7f"
-  steelblue: "#4682b4"
-  tan: "#d2b48c"
-  teal: "#008080"
-  thistle: "#d8bfd8"
-  tomato: "#ff6347"
-  turquoise: "#40e0d0"
-  violet: "#ee82ee"
-  wheat: "#f5deb3"
-  white: "#ffffff"
-  whitesmoke: "#f5f5f5"
-  yellow: "#ffff00"
-  yellowgreen: "#9acd32"
-
-colorToHex = (color) ->
-  return color if not color
-  return color.toLowerCase() if color.substr(0, 1) is "#" or color is "transparent"
-  return colorNameToHexMap[color.toLowerCase()] if color of colorNameToHexMap
-  digits = /(.*?)rgb(a)?\((\d+), (\d+), (\d+)(, (\d+))?\)/.exec(color)
-  if digits?.length is 8
-    red = parseInt(digits[3])
-    green = parseInt(digits[4])
-    blue = parseInt(digits[5])
-    opacity = parseInt(digits[7])
-    if opacity == 0
-      return "transparent"
-    return rgbToHex(red, green, blue)
-  return undefined
-
-Ember.Widgets.ColorPickerComponent = Ember.Component.extend
+###*
+ * ColorPickerComponent
+ *
+ * This is the whole color picker component. It is the interface connecting with
+ * users.
```

Comment:
I think it wasn't good at the beginning. I would like to emphasize that it is the component but it does have sub-components where most of the logics are in the dropdown component.